### PR TITLE
Target es6, use Object.assign

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Simple clean up PR that covers a few things:
- Changes TS config to target es6 instead of es5
- Uses `Object.assign` instead of the util lib in the auth library
- Switch a few modules over to es6 style imports

### CODE REVIEW

**Comment by lukesneeringer** [comment]
Does the `import` statement work in Node 4? I thought it did not.

**Comment by ofrobots** [comment]
@lukesneeringer note that this is in typescript now.

File: test/test.auth.ts [hunk 1]
```diff
@@ -15,8 +15,9 @@ import * as assert from 'power-assert';
 import * as async from 'async';
 import * as nock from 'nock';
 import utils from './utils';
-let googleapis = require('../');
-let googleauth = require('../lib/googleauth.js');
+import googleauth from '../lib/googleauth';
+const googleapis = require('../');
```
Comment (jmdobry) [inline test/test.auth.ts]: Why can't `import` be used here?

Comment (JustinBeckwith) [inline test/test.auth.ts]: Right now googleapis.ts doesn't currently export an es6 module.  That change is very involved, and best kept separate from this one.

**Comment by ofrobots** [comment]
Re: target es6: that wouldn't necessarily work with Node 4 as it doesn't have complete ES6 support.

<insert lament about TypeScript offering implementation levels rather than spec levels in their target options>

**Comment by jmdobry** [comment]
A good example would be array destructing, can't use it in our samples :(

**Comment by JustinBeckwith** [comment]
I am counting on the Travis tests that cover Node 4 here :)  Are you ok with me targeting es6?

**Comment by ofrobots** [comment]
Given that the code coverage is at 79%, are we confident that we can trust the tests on this matter?

**Comment by JustinBeckwith** [comment]
Some good news here is that the scripts/ dir, which contains the generator code, is by far the biggest culprit here:

image

I can work on getting better coverage for discovery.ts before moving on.

**Pull request merged**